plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -352,6 +360,62 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(
+            log.info("Performing {} bulk actions on table {}", request.requests().size(), request.getIndices());
+
+            List<DataObjectResponse> responses = new ArrayList<>();
+
+            long startNanos = System.nanoTime();
+
+            for (DataObjectRequest dataObjectRequest : request.requests()) {
+                try {
+                    if (dataObjectRequest instanceof PutDataObjectRequest) {
+                        responses
+                            .add(
+                                putDataObjectAsync((PutDataObjectRequest) dataObjectRequest, executor, isMultiTenancyEnabled)
+                                    .toCompletableFuture()
+                                    .join()
+                            );
+                    } else if (dataObjectRequest instanceof UpdateDataObjectRequest) {
+                        responses
+                            .add(
+                                updateDataObjectAsync((UpdateDataObjectRequest) dataObjectRequest, executor, isMultiTenancyEnabled)
+                                    .toCompletableFuture()
+                                    .join()
+                            );
+                    } else if (dataObjectRequest instanceof DeleteDataObjectRequest) {
+                        responses
+                            .add(
+                                deleteDataObjectAsync((DeleteDataObjectRequest) dataObjectRequest, executor, isMultiTenancyEnabled)
+                                    .toCompletableFuture()
+                                    .join()
+                            );
+                    }
+                } catch (CompletionException e) {
+                    if (dataObjectRequest instanceof PutDataObjectRequest) {
+                        responses.add(new PutDataObjectResponse.Builder().id(dataObjectRequest.id()).failed(true).build());
+                    } else if (dataObjectRequest instanceof UpdateDataObjectRequest) {
+                        responses.add(new UpdateDataObjectResponse.Builder().id(dataObjectRequest.id()).failed(true).build());
+                    } else if (dataObjectRequest instanceof DeleteDataObjectRequest) {
+                        responses.add(new DeleteDataObjectResponse.Builder().id(dataObjectRequest.id()).failed(true).build());
+                    }
+                    log.error("Error in bulk operation for id {}: {}", dataObjectRequest.id(), e.getCause().getMessage(), e.getCause());
+                }
+            }
+            long endNanos = System.nanoTime();
+            long tookMillis = TimeUnit.NANOSECONDS.toMillis(endNanos - startNanos);
+
+            log.info("Bulk action complete for {} items, took {} ms", responses.size(), tookMillis);
+            return new BulkDataObjectResponse(responses.toArray(new DataObjectResponse[0]), tookMillis, NO_INGEST_TOOK);
```

Comment:
Just curious to know what is `NO_INGEST_TOOK`?

Reply:
The `BulkResponse` has 3 fields, the array of `BulkItemResponse` objects, how long the operation took, and in some special cases how long ingestion took.  Since we're not ingesting that field stays at its default, which here is statically imported from here, basically indicating that data is not relevant:
https://github.com/opensearch-project/OpenSearch/blob/0363aa79c434343dc7f5b8df9f77b2d4652f4c7e/server/src/main/java/org/opensearch/action/bulk/BulkResponse.java#L70

Reply:
The `BulkResponse` has a 2-arg and 3-arg constructor, I could do the same and hide this implementation detail... probably a better option.